ingestion-beam/src/test/java/com/mozilla/telemetry/decoder/ParseUriTest.java

```diff
@@ -317,36 +327,62 @@ private static Map<String, String> joinCases(List<Map<String, String>> cases) {
           "v8-3", "\"ping_version\":\"v8-3\",\"funnelcake\":\"3\""), //
       VALID_V8_DIMENSIONS));
 
+  private static final Map<String, String> VALID_V9_CASES = joinCases(Arrays.asList(//
+      ImmutableMap.of("v9", "\"ping_version\":\"v9\"", //
+          "v9-3", "\"ping_version\":\"v9-3\",\"funnelcake\":\"3\""), //
+      VALID_V9_DIMENSIONS));
+
+  private static final Map<String, String> VALID_V10_CASES = joinCases(Arrays.asList(//
+      ImmutableMap.of("v10", "\"ping_version\":\"v10\"", //
+          "v10-3", "\"ping_version\":\"v10-3\",\"funnelcake\":\"3\""), //
+      VALID_V10_DIMENSIONS));
+
   @Test
   public void testStubUri() {
     final List<String> input = Streams.stream(Iterables.concat(//
         VALID_V6_CASES.keySet(), //
         VALID_V7_CASES.keySet(), //
         VALID_V8_CASES.keySet(), //
+        VALID_V9_CASES.keySet(), //
+        VALID_V10_CASES.keySet(), //
         Arrays.asList(//
             "v6/" + String.join("/", Collections.nCopies(35, " ")), //
             "v7/" + String.join("/", Collections.nCopies(36, " ")), //
             "v8/" + String.join("/", Collections.nCopies(38, " ")), //
-            "", "v6", "v7", "v8")))
-        .map(v -> "{\"attributeMap\":{\"message_id\":\"00000000001\",\"uri\":\"/stub/" + v
+            "v9/" + String.join("/", Collections.nCopies(40, " ")), //
+            "v10/" + String.join("/", Collections.nCopies(42, " ")), //
+            "v6", "v7", "v8", "v9", "v10", // UnexpectedPathElementsException
+            "", "v1", "v61", "v21" // UnknownPingVersionException
```

Comment:
Thanks for adding comments, it's not immediately obvious what's going on here ☠️